app/models/audio_file.rb

```diff
@@ -48,4 +35,29 @@ class AudioFile < BaseModel
   def fixerable_final?
     [VALID, COMPLETE, TRANSFORMING, TRANSFORM_FAILED, TRANSFORMED].include? status
   end
+
+  def compliant_with_template?
+    audio_errors.nil?
+  end
+
+  def validate_on_template
+    # don't bother checking template for audio file that's not yet ready
+    # note: this does mean that files previously deemed invalid by template
+    # will never make it past this line & will stay INVALID.
+    return if [NOTFOUND, VALIDATING, INVALID, FAILED].include?(status)
```

Comment:
Is there a reason not to re-run the checks when `INVALID`?  Just to save some time?